allure-web/src/plugins/timeline/TimelineModel.js

```diff
@@ -14,4 +14,40 @@ export default class TimelineModel extends Model {
         }
         return this.allTestcases;
     }
-}
+
+    getTestcases(host) {
+        const startTime = this.get('time').start;
+        return host.children.reduce((testcases, thread) =>
+            testcases.concat(thread.children.map(({time, uid, name, status}) => {
+                return {
+                    uid, name, status,
+                    start: time.start - startTime,
+                    stop: time.stop - startTime,
+                    thread: thread.name
+                };
+            })),
+            []
+        );
+    }
+
+    getFilteredData(minDuration){
+        var data = this.get('children');
+        var total = this.get('statistic').total;
+        this.data = Object.assign({}, {
+            children: data.map(host => {
+                return Object.assign({}, host, { children: host['children'].map(thread => {
+                    return Object.assign({}, thread, { children: thread['children'].filter(testCase => {
+                            if (testCase.time.duration >= minDuration){
+                                return true;
+                            } else {
+                                total--;
+                                return false;
+                            }
+                        })
+                    });
+                }).filter(d=> {return d['children'].length;})});
```

Comment:
`d.children.length`?
String keys are used only for dynamic lookup, like `d[keyName]`